plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Transform/Inline.hs

```diff
@@ -67,11 +68,33 @@ newtype TermEnv name uni fun a =
 -- See Note [Differences from PIR inliner] 1
 newtype Subst name uni fun a = Subst { _termEnv :: TermEnv name uni fun a }
     deriving stock (Generic)
-    deriving (Semigroup, Monoid) via (GenericSemigroupMonoid (Subst name uni fun a))
+    deriving newtype (Semigroup, Monoid)
 
 makeLenses ''TermEnv
 makeLenses ''Subst
 
+data VarInfo name uni fun ann = VarInfo
+  { _varBinders :: [name]
+  , _varRhs     :: Term name uni fun ann
+  , _varRhsBody :: InlineTerm name uni fun ann
```

Comment:
maybe mention `InlineTerm` is needed instead of `Term` and point to the `Dupable` note.